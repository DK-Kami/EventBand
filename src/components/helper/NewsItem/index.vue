<template>
	<v-card
		class="my-4"
	>
		
		<v-toolbar 
			v-if="showOrganizationName"
			color="secondary"
		>
			<v-toolbar-title class="white--text"> {{ organization.name }} </v-toolbar-title>
		</v-toolbar>

		<v-card-title>
			<v-flex xs12>
				<div> 
					Topic: <span class="grey--text text--darken-2"> {{ news.title }} </span>
				</div>
			</v-flex>
		
		</v-card-title>
		<v-card-text>
			<v-layout>
				<span class="grey--text text--darken-3"> {{ news.text }} </span>
			</v-layout>
		</v-card-text>
		<v-divider></v-divider>
		<v-card-subtitle class="pa-0 px-4"> 
			<v-flex xs12>
				<span class="subtitle-2">
					Created at: {{ dateAt }} ({{timeAt}})
				</span>
			</v-flex>
		</v-card-subtitle>
	</v-card>
</template>

<script>
export default {
	name: 'NewsItem',

	props: {
		showOrganizationName: {
			type: Boolean,
			default: true,
		},
		news: {
			type: Object,
			default: () => ({
				uuid: '01f77df9-d732-4d2a-907d-7ad4edadf7e0',
				title: 'News Title',
				text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ea non cupiditate praesentium inventore voluptatem quas?',
				createdAt: '2020-05-21T13:51:35.073Z',
			}),
		},
		organization: {
			type: Object,
			default: () => ({
				uuid: 'bfc0de35-5789-4746-aef3-aa5bc393836a',
				logo: 'no-homo.jpg',
				name: 'Org name',
			}),
		}
	},

	computed: {
		dateAt() {
			return this.news.createdAt.split('T')[0];
		},

		timeAt() {
			return this.news.createdAt.split('T')[1].split('Z')[0].split('.')[0];
		},
	},

}
</script>