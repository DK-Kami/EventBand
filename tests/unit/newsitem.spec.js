import NewsItem from '@/components/helper/NewsItem';
import { mount, shallowMount } from './index';

describe('News item - компонент для вывода новостей', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(NewsItem);
	});

	it ('Делаем снапшот', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Является компонентом Vue', () => {
		expect(wrapper.isVueInstance()).toBeTruthy();
	});
	
	it ('Имеет toolbar заголовок', () => {
		expect(wrapper.find('.v-toolbar__title').isVisible()).toBe(true);
	});

	it('Не имеет toolbar заголовок', () => {
		wrapper = shallowMount(NewsItem, {
			store: false,
			propsData: {
				showOrganizationName: false,
			},
		});
		
		expect(wrapper.contains('v-toolbar-title')).toBe(false);
	});

	it ('Имеет toolbar заголовок c текстом "Org name"', () => {
		wrapper = shallowMount(NewsItem);
			
		expect(wrapper.find('.white--text').html()).toContain('Org name');
	});

	it ('Имеет toolbar загловок с текстом "Unique name"', () => {
		wrapper = shallowMount(NewsItem, {
			store: false,
			propsData: {
				organization: {
					uuid: 'bfc0de35-5789-4746-aef3-aa5bc393836a',
					logo: 'no-homo.jpg',
					name: 'Unique name',
				},
			}
		});

		expect(wrapper.find('.white--text').html()).toContain('Unique name');
	});

	it ('Сard имеет заголовок', () => {
		expect(wrapper.contains('.v-card__title')).toBeTruthy();
	});

	it ('Сard имеет подзаголовок', () => {
		expect(wrapper.contains('.v-card__subtitle')).toBeTruthy();
	});

	it ('Сard имеет текст', () => {
		expect(wrapper.contains('.v-card__text')).toBeTruthy();
	});
	
});
