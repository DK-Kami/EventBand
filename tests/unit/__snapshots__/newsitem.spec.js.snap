// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News item - компонент для вывода новостей Делаем снапшот 1`] = `
<div class="my-4 v-card v-sheet theme--light">
  <header class="v-sheet v-sheet--tile theme--light v-toolbar secondary" style="height: 64px;">
    <div class="v-toolbar__content" style="height: 64px;">
      <div class="v-toolbar__title white--text"> Org name </div>
    </div>
  </header>
  <div class="v-card__title">
    <div class="flex xs12">
      <div>
        Topic: <span class="grey--text text--darken-2"> News Title </span></div>
    </div>
  </div>
  <div class="v-card__text">
    <div class="layout"><span class="grey--text text--darken-3"> Lorem ipsum dolor sit amet consectetur adipisicing elit. Ea non cupiditate praesentium inventore voluptatem quas? </span></div>
  </div>
  <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
  <div class="v-card__subtitle pa-0 px-4">
    <div class="flex xs12"><span class="subtitle-2">
				Created at: 2020-05-21 (13:51:35)
			</span></div>
  </div>
</div>
`;
